### ISSUE TITLE
Wrong kotlin code generated for SQL Server stored procedures

### ISSUE THREAD

**Issue by thomasobrien99** [jOOQ/jOOQ#13804]
### Expected behavior

Generated Routines for stored procedures compile

### Actual behavior

Generated Routines for stored procedures can fail compilation with: `Unresolved reference: syntheticReturnParameter`

### Steps to reproduce the problem

I'm thinking that this is a bug rather than a configuration issue, but if it's not feel free to close/re-direct me to stack overflow.

Running generate*Jooq is creating Routine files with:
```
...
init {
        syntheticReturnParameter = RETURN_VALUE
        addInParameter(Foobar)
        ...
    }
...
```
where `syntheticReturnParameter` is an unresolved reference:
<img width="1053" alt="image" src="https://user-images.githubusercontent.com/13715890/178334923-1973e7fc-10e0-4606-9057-1ef1c6b0c50e.png">

- If the problem relates to code generation, please post your code generation configuration:

This using the gradle jooq plugin: `id("nu.studer.jooq") version "7.1.1"`

```
create("ourgeneratedcode") {
      generateSchemaSourceOnCompilation.set(false)

      jooqConfiguration.apply {
        logging = Logging.INFO
        generator.apply {
          name = "org.jooq.codegen.KotlinGenerator"
          strategy.apply {
            name = "org.jooq.codegen.KeepNamesGeneratorStrategy"
          }
          database.apply {
            name = "org.jooq.meta.sqlserver.SQLServerDatabase"
            inputCatalog = "experiment"
            schemata.addAll(
              arrayOf(
                org.jooq.meta.jaxb.SchemaMappingType()
                  .withInputSchema("dbo"),
                org.jooq.meta.jaxb.SchemaMappingType()
                  .withInputSchema("otherSchema"),
              ),
            )
            isIncludeIndexes = false
            properties.add(
              Property().apply {
                key = "locations"
                value = "src/main/resources/ourMigrations"
              },
            )
          generate.apply {
            isDeprecated = false
            isRecords = true
            isPojos = true
            isPojosEqualsAndHashCode = false
          }
          target.apply {
            packageName = "com.ourPackageName"
            directory =
              "src/main/kotlin/generated/ourGeneratedCodeLocation"
          }
        }
      }
    }
```

### Versions

- jOOQ: 3.16.4
- Java: 17
- Database (include vendor): SQL Server
- OS: `iOS 12.4`
- JDBC Driver (include name if unofficial driver): `com.microsoft.sqlserver.jdbc.SQLServerDriver`

**Comment by lukaseder** [comment]
Sorry for the delay in this. I'll look into this right away now.

**Comment by lukaseder** [comment]
The problem is that `syntheticReturnParameter` doesn't work like a property as there isn't any getter, only a setter. This should be an easy fix. The generated code should produce:

```kotlin
setSyntheticReturnParameter(RETURN_VALUE)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.3 (https://github.com/jOOQ/jOOQ/issues/13864), and 3.16.9 (https://github.com/jOOQ/jOOQ/issues/13865)

**Comment by lukaseder** [comment]
The workaround is to use the `JavaGenerator`, at least for generated routines.
